workshop-destroy.py

```diff
@@ -4,21 +4,43 @@
 import yaml
 import os
 import shutil
+import boto3
+import botocore
 
 argparse = argparse.ArgumentParser()
 argparse.add_argument('--dir', help="Workshop directory", required=True)
 args = argparse.parse_args()
 workshop_home = args.dir
 
 # Open and parse configuration file
-with open( os.path.join(workshop_home, "workshop.yaml"), 'r') as yaml_file:
+with open(os.path.join(workshop_home, "workshop.yaml"), 'r') as yaml_file:
     try:
         config = yaml.safe_load(yaml_file)
     except yaml.YAMLError as exc:
         print(exc)
 
-terraform_staging=os.path.join(args.dir, ".terraform_staging" )
-docker_staging=os.path.join(args.dir, ".docker_staging")
+
+def check_login():
+    boto3.setup_default_session(profile_name=(config['workshop']['core']['profile']))
+    sts = boto3.client('sts')
+    try:
+        sts.get_caller_identity()
+        return True
+    except botocore.exceptions.UnauthorizedSSOTokenError:
+        return False
+    except botocore.exceptions.ClientError:
+        return False
+
+
+if (config['workshop']['core']['cloud_provider']) == 'aws':
```

Comment:
Sames as above. Nice solution to the problem. Can we add likewise check to other clouds?